PurchasesCoreSwiftTests/LocalReceiptParsing/DataConverters/ISO3601DateFormatterTests.swift

```diff
@@ -15,10 +15,26 @@ class ISO3601DateFormatterTests: XCTestCase {
                                             minute: 36,
                                             second: 40)
         let date = Calendar.current.date(from: dateComponents)
-        guard let dateBytes = "2020-07-14T19:36:40+0000".data(using: .ascii) else { fatalError() }
+        guard let dateBytes = "2020-07-14T19:36:40Z".data(using: .ascii) else { fatalError() }
         expect(ISO3601DateFormatter.shared.date(fromBytes: ArraySlice(dateBytes))) == date
     }
 
+    func testDateWithMillisecondsFromBytesReturnsCorrectValueIfPossible() {
+        let timeZone = TimeZone(identifier: "UTC")
+        let dateComponents = DateComponents(timeZone: timeZone,
+                                            year: 2020,
+                                            month: 7,
+                                            day: 14,
+                                            hour: 19,
+                                            minute: 36,
+                                            second: 40,
+                                            nanosecond: 202_000_000)
+        let date = Calendar.current.date(from: dateComponents)
+        guard let dateBytes = "2020-07-14T19:36:40.202Z".data(using: .ascii) else { fatalError() }
+        let receivedDate = ISO3601DateFormatter.shared.date(fromBytes: ArraySlice(dateBytes))
+        expect(receivedDate!.timeIntervalSince1970).to(beCloseTo(date!.timeIntervalSince1970))
```

Comment:
We can't compare same way we do in the other test because of the precision issue you mentioned to me in private?

Reply:
yeah, floating point precision is a killer here.

Reply:
really, we should be doing this on any and all floating point comparisons. I can update the other one.